Source/ACE.Server/Command/Handlers/DeveloperCommands.cs

```diff
@@ -2135,6 +2138,34 @@ public static void HandleRequireComps(Session session, params string[] parameter
                     session.Network.EnqueueSend(new GameMessageSystemChat("You can no longer cast spells without components.", ChatMessageType.Broadcast));
                     break;
             }
+        }
+        /// <summary>
+        /// This is to add spells to items (whether loot or quest generated).  For making weapons to check damage from pcaps or other sources
+        /// </summary>
+        [CommandHandler("additemspell", AccessLevel.Developer, CommandHandlerFlag.RequiresWorld, 1, "/additemspell <spell id> - adds a spell to the last appraised item. Ex /additemspell 6089 - adds Legendary Bloodthirst")]
+        public static void HandleAddItemSpell (Session session, params string[] parameters)
+        {
+            var obj = CommandHandlerHelper.GetLastAppraisedObject(session);
+            var spellId = Convert.ToUInt32(parameters[0]);
+            //Add Spell Check
+
+            //Check to see if Spell ID Parameter is given
+            //if (spellId == null)
+            //    session.Network.EnqueueSend(new GameMessageSystemChat("Please Speficy SpellID.", ChatMessageType.Broadcast));
+            //return;
+
+            //Check to see if Spell ID is valid spell
+            var spell = new Spell(spellId, true);
+            if (spell.NotFound == true)
+            {
+                session.Network.EnqueueSend(new GameMessageSystemChat("SpellID is not found", ChatMessageType.Broadcast));
+                return;
+            }
+            obj.Biota.GetOrAddKnownSpell((int)spellId, obj.BiotaDatabaseLock, obj.BiotaPropertySpells, out _);
+
+
+
+
         }
     }
 }
```

Comment:
fixed the above crash, and added a result message:

```
        /// <summary>
        /// This is to add spells to items (whether loot or quest generated).  For making weapons to check damage from pcaps or other sources
        /// </summary>
        [CommandHandler("additemspell", AccessLevel.Developer, CommandHandlerFlag.RequiresWorld, 1, "/additemspell <spell id> - adds a spell to the last appraised item. Ex /additemspell 6089 - adds Legendary Bloodthirst")]
        public static void HandleAddItemSpell(Session session, params string[] parameters)
        {
            var obj = CommandHandlerHelper.GetLastAppraisedObject(session);

            if (!int.TryParse(parameters[0], out var spellId))
                return;

            // ensure valid spell id
            var spell = new Entity.Spell(spellId);

            if (spell.NotFound)
            {
                session.Network.EnqueueSend(new GameMessageSystemChat("SpellID is not found", ChatMessageType.Broadcast));
                return;
            }

            obj.Biota.GetOrAddKnownSpell(spellId, obj.BiotaDatabaseLock, obj.BiotaPropertySpells, out var spellAdded);

            var msg = spellAdded ? "added to" : "already on";

            session.Network.EnqueueSend(new GameMessageSystemChat($"{spell.Name} ({spell.Id}) {msg} {obj.Name}", ChatMessageType.Broadcast));
        }
```

Reply:
Added Try parse for converting spellId to Uint. Added other messages to let you know if spell is already on item per gmriggs